source/slang/slang-emit-cpp.cpp

```diff
@@ -2589,34 +2589,43 @@ void CPPSourceEmitter::emitModuleImpl(IRModule* module)
 
                                 Slang::ComPtr<ISlangBlob> blob;
                                 if (SLANG_FAILED(result.getBlob(blob)))
                                 {
-                                    SLANG_UNEXPECTED("No blob to emit");
-                                    return;
+                                    StringBuilder builder;
+                                    sink->diagnoseRaw(Severity::Error, UnownedStringSlice
+                                        ("Slang heterogeneous error: No blob to emit\n"));
+                                    m_writer->emit("size_t __");
+                                    m_writer->emit(entryPointName);
+                                    m_writer->emit("Size = 0;\n");
+                                    m_writer->emit("unsigned char __");
+                                    m_writer->emit(entryPointName);
+                                    m_writer->emit("[];\n");
                                 }
 
-                                auto ptr = (const unsigned char*)blob->getBufferPointer();
-
-                                m_writer->emit("size_t __");
-                                m_writer->emit(entryPointName );
-                                m_writer->emit("Size = ");
-                                m_writer->emitInt64(blob->getBufferSize());
-                                m_writer->emit(";\n");
-
-                                m_writer->emit("unsigned char __");
-                                m_writer->emit(entryPointName );
-                                m_writer->emit("[] = {");
-                                for (unsigned int i = 0; i < blob->getBufferSize() - 1; i++) {
-                                    m_writer->emitUInt64(ptr[i]);
-                                    m_writer->emit(", ");
+                                else {
+                                    auto ptr = (const unsigned char*)blob->getBufferPointer();
+
+                                    m_writer->emit("size_t __");
+                                    m_writer->emit(entryPointName);
+                                    m_writer->emit("Size = ");
+                                    m_writer->emitInt64(blob->getBufferSize());
+                                    m_writer->emit(";\n");
+
+                                    m_writer->emit("unsigned char __");
+                                    m_writer->emit(entryPointName);
+                                    m_writer->emit("[] = {");
+                                    for (unsigned int i = 0; i < blob->getBufferSize() - 1; i++) {
+                                        m_writer->emitUInt64(ptr[i]);
+                                        m_writer->emit(", ");
+                                    }
+                                    m_writer->emitUInt64(ptr[blob->getBufferSize() - 1]);
+                                    m_writer->emit("};\n");
```

Comment:
As a suggestion it might be simpler to have a function that handled this writing out that could be used for either case. It would need a bit of special case code around the 0 size (for failed getting of a blob say), but overall should cut down the code. Say something like...

```
void writeBlob(const void* data, size_t dataCount, const char* entryPointName, SourceWriter* writer)
```

That the function would have to test if the size is 0, and still output [1] for the data.

You might also want to output a cr every n bytes - as seen in HexDumpUtil::dumpSourceBytes.